vrs/vrs-annotation-batch-wrapper.py

```diff
@@ -0,0 +1,109 @@
+TODO: add possiblity to zip output 
+usage: python3 vrs-annotation-batch-wrapper.py --billing-project gnomad-vrs \
+    --tmp-dir gnomad-vrs \
+    --bucket-mount gnomad-vrs \
+    --image us-central1-docker.pkg.dev/broad-mpg-gnomad/ga4gh-vrs/marten-vrs-image:0221 \
+    --vcf-in vcf-10k-sharded-0228.vcf.bgz \
+    --vcf-out gs://gnomad-vrs/vcf-test-0302.vcf.bgz
+"""
+
+import argparse
+import datetime
+import errno
+import logging
+import os
+import sys
+
+import hail as hl
+import hailtop.batch as hb
+from batch_utils import init_job
+
+
+def init_job_with_gcloud(
+    batch,
+    name: str = None,
+    image: str = None,
+    cpu: float = None,
+    memory: float = None,
+    disk_size: float = None,
+    mount: str = None,
+    mount_to: str = None,
+):
+    """
+    Create job and initialize glcoud authentication and gsutil commands.
+    Wraps Ben Weisburd's init_job (https://github.com/macarthur-lab/methods/blob/master/batch/batch_utils.py#L101) with additional gcloud steps.
+    Parameters passed through to init_job (descriptions taken from github documentation for the function):
+        :param batch: Batch object
+        :param name: job label which will show up in the Batch web UI
+        :param image: docker image name (eg. "weisburd/image-name@sha256:aa19845da5")
+        :param cpu: number of CPUs (between 0.25 to 16)
+        :param memory: amount of RAM in Gb (eg. 3.75)
+        :param disk_size: amount of disk in Gb (eg. 50)
+        :return: new job object
+    """
+    job = init_job(batch, name, image, cpu, memory, disk_size)
+    job.command(f"gcloud -q auth activate-service-account --key-file=/gsa-key/key.json")
+    job.command(f"curl -sSL broad.io/install-gcs-connector | python3")
+    return job
+
+
+def main(args):
+
+    backend = hb.ServiceBackend(args.billing_project, args.tmp_dir)
+
+    batch001 = hb.Batch(name="fuse_batch", backend=backend)
+
+    index_list = [str(yi).zfill(3) for yi in range(100)]
+    # TODO: this only works for sharded VCFs with 100 partitions
+    # Needs fixing and edits in future to be more universifiable
+
+    for vcf_index in index_list:
+
+        new_job = init_job_with_gcloud(
+            batch=batch001, name=f"VCF_{vcf_index}_job", image=args.image
+        )
+        new_job.cloudfuse(f"{args.bucket_mount}", "/local-seqrepo")
+        # NOTE: the script gs://gnomad-vrs/marten-vrs-annotation.py has minor edits from 'vcf_annotation.py' from the GA4GH Team
+        # namely, 'click' functionality for arguments has been changed to ArgParser
+        # and some custom errors they included could not be updated
+        # THESE WILL BE CHANGED IN THE FUTURE ! 
+        new_job.command(
+            f"python3 /local-seqrepo/marten-vrs-annotation.py --vcf-in /local-seqrepo/{args.vcf_in}/part-{vcf_index}*.bgz --vcf-out vcf-output-{vcf_index}.vcf --seqrepo-path /local-seqrepo/seqrepo/2018-11-26/"
```

Comment:
should use their script directly rather than an edited copy

Reply:
noted - I just Click and that Error importing step fixed, so I should be good to use that in the future!